src/main/webapp/app/shared/metis/posting-content/posting-content.components.ts

```diff
@@ -56,14 +56,33 @@ export class PostingContentComponent implements OnInit, OnDestroy {
                 const referencedId = this.content!.substring(patternMatch.startIndex + 1, patternMatch.endIndex); // e.g. post id 6
-                const referenceStr = this.content!.substring(patternMatch.startIndex, patternMatch.endIndex); // e.g. '#6'
-
-                // if the referenced Id is within the currently loaded posts, we can create the context-specific link to that post
-                // by invoking the respective metis service methods for link and query params and passing the post object;
-                // if not, we do not want to fetch the post from the DB and rather always navigate to the course discussion page with the referenceStr as search text
-                const referencedPostInLoadedPosts = this.currentlyLoadedPosts.find((post: Post) => post.id! === +referencedId);
-                const linkToReference = this.metisService.getLinkForPost(referencedPostInLoadedPosts);
-                const queryParams = referencedPostInLoadedPosts ? this.metisService.getQueryParamsForPost(referencedPostInLoadedPosts) : ({ searchText: referenceStr } as Params);
+                const referenceType = patternMatch.referenceType;
+                let referenceStr; // e.g. '#6'
+                let linkToReference;
+                let attachmentToReference;
+                let queryParams;
+                if (ReferenceType.POST === referenceType) {
+                    // if the referenced Id is within the currently loaded posts, we can create the context-specific link to that post
+                    // by invoking the respective metis service methods for link and query params and passing the post object;
+                    // if not, we do not want to fetch the post from the DB and rather always navigate to the course discussion page with the referenceStr as search text
+                    const referencedPostInLoadedPosts = this.currentlyLoadedPosts.find((post: Post) => post.id! === +referencedId);
+                    referenceStr = this.content!.substring(patternMatch.startIndex, patternMatch.endIndex);
+                    linkToReference = this.metisService.getLinkForPost(referencedPostInLoadedPosts);
+                    queryParams = referencedPostInLoadedPosts ? this.metisService.getQueryParamsForPost(referencedPostInLoadedPosts) : ({ searchText: referenceStr } as Params);
+                } else if (
+                    ReferenceType.LECTURE === referenceType ||
+                    ReferenceType.PROGRAMMING === referenceType ||
+                    ReferenceType.MODELING === referenceType ||
+                    ReferenceType.QUIZ === referenceType ||
+                    ReferenceType.TEXT === referenceType ||
+                    ReferenceType.FILE_UPLOAD === referenceType
+                ) {
+                    referenceStr = this.content!.substring(this.content?.indexOf(')', patternMatch.startIndex)! + 1, patternMatch.endIndex - (referenceType.length + 3));
```

Comment:
Could you please add some comments and an example to understand the string magic? :D e.g. why do you add 3, ...